pilot/pkg/xds/cds.go

```diff
@@ -47,13 +47,13 @@ func cdsDiscoveryResponse(response []*cluster.Cluster, noncePrefix string) *disc
 }
 
 func (s *DiscoveryServer) pushCds(con *Connection, push *model.PushContext, version string) error {
-	// TODO: Modify interface to take services, and config instead of making library query registry
 	pushStart := time.Now()
+	defer cdsPushTime.Record(time.Since(pushStart).Seconds())
```

Comment:
This doesn't work. `time.Since(pushStart)` is evaluated immediately, not in the defer call.
This must be wrapped into a func:
```
defer func() { cdsPushTime.Record(time.Since(pushStart).Seconds()) }
```
Same for the other defers below.

Reply:
thanks, you are right. The param is calculated first.

Reply:
https://play.golang.org/p/d7AcM1fAImX